spatial/r3/mat.go

```diff
@@ -197,3 +197,20 @@ func (m *Mat) Det() float64 {
 	detc := m.At(1, 0)*m.At(2, 1) - m.At(1, 1)*m.At(2, 0)
 	return a*deta - b*detb + c*detc
 }
+
+// Skew sets the receiver to the 3×3 skew symmetric matrix
+// (right hand system) of v.
+//                  ⎡ 0 -z  y⎤
+//  Skew({x,y,z}) = ⎢ z  0 -x⎥
+//                  ⎣-y  x  0⎦
+func (m *Mat) Skew(v Vec) {
```

Comment:
What is the utility of this matrix? If it's to compute its product with a vector, then why not do it more efficiently with `Cross`? Are there other uses?

Reply:
I guess the answer is "composition".

Reply:
One use I have here lying around is computing a rotation tensor from a quaternion

```go
// RotationFromQuat stores the rotation matrix from a quaternion (must be normalized).
//  [R] = q.Real*q.Real * [E] - dot([q],[q])*[E] + 2*[q]*[q]ᵀ + 2*q.Real * skew([q])
// where
//  [q] = {q.Imag, q.Jmag, q.Kmag}
// and
//  [E]=Eye()
func (m *Mat) RotationFromQuat(q quat.Number) {
	qv := r3.Vec{X: q.Imag, Y: q.Jmag, Z: q.Kmag}
	qs := Skew(qv)
	q01 := Identity()
	q01.Scale(q.Real*q.Real, q01)

	qd := Identity()
	qd.Scale(r3.Dot(qv, qv), qd)

	qs.Scale(2*q.Real, qs)

	qv = r3.Scale(2, qv)
	m.Prod(qv, qv) // m = 2*[q]*[q]ᵀ
	m.Add(m, q01)  // m += q.Real*q.Real * [E]
	m.Add(m, qd)   // m += dot([q],[q])*[E]
	m.Add(m, qs)   // m += 2*q.Real * skew([q])
}
```

The literature I follow for rigid body dynamics uses 3x3 matrices to store angular velocities, though I've still have not gotten as far as to using them in that form.